### PR TITLE
sets aria-label attribute for Path and Coordinates

### LINKED ISSUE #18: set ariaLabels on shapes
dependent on this issue: https://github.com/d3plus/d3plus-shape/issues/78

`ariaLabel` should be based on the following formulas:

- [ ] Path: `${drawLabel}, ${colorScale value}.`
- [ ] Coordinate: `${drawLabel}, ${radius value}.`

### PR DESCRIPTION
**Pull request by rbaheti**
### Description

- Sets aria-label attribute for Path using formula: ${drawLabel}, ${colorScale value}.
- Sets aria-label attribute for Coordinate using formula: ${drawLabel}, ${radius value}.
closes d3plus/d3plus-geomap#18

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist

- [ ] I have documented all new methods.
- [ ] I have written tests for all new methods/functionality.
- [ ] I have written examples for all new methods/functionality.

### CODE REVIEW

File: src/Geomap.js [hunk 1]
```diff
@@ -58,8 +58,10 @@ export default class Geomap extends Viz {
 
     this._shape = constant("Circle");
     this._shapeConfig = assign(this._shapeConfig, {
+      ariaLabel: (d, i) => `${this._drawLabel(d, i)}, ${this._pointSize(d, i)}`,
       hoverOpacity: 1,
       Path: {
+        ariaLabel: (d, i) => `${this._drawLabel(d, i)}, ${this._colorScale(d, i)}.`,
```
Comment (davelandry) [inline src/Geomap.js]: This needs a catch, similar to the one in d3plus-network regarding size, if the `colorScale` is not defined (only show the label in this case).

**Review by davelandry — CHANGES_REQUESTED** [review]

File: src/Geomap.js [hunk 2]
```diff
@@ -58,8 +58,13 @@ export default class Geomap extends Viz {
 
     this._shape = constant("Circle");
     this._shapeConfig = assign(this._shapeConfig, {
+      ariaLabel: (d, i) => `${this._drawLabel(d, i)}, ${this._pointSize(d, i)}`,
       hoverOpacity: 1,
       Path: {
+        ariaLabel: (d, i) => {
+          const validColorScale = this._colorScale ? `, ${this._colorScale(d, i)}` : "";
+          return `${this._drawLabel(d, i)}, ${validColorScale}.`;
```
Comment (davelandry) [inline src/Geomap.js]: remove `", "` from the return statement, as it is being added by the validColorScale.

for example, if a shape named "Alpha" does not have a validColorScale, the label is rendered as `"Alpha, ."`

**Review by davelandry — CHANGES_REQUESTED** [review]

**Review by davelandry — APPROVED** [review]

**Comment by yemoku** [comment]
Hello,

Is there a complete sample/example for this, as in providing a caption for each state/shape on the map?

Thanks

**Comment by davelandry** [comment]
@yemoku this code overrides the aria-label on the `<path>` elements created by a map:

```js
.config({
  shapeConfig: {
    Path: {
      ariaLabel: (d, i) => "my custom label"
    }
  }
})
```

**Comment by yemoku** [comment]
Thanks for responding. tried that but no joy. My full code: to even put the same label in all the shapes/paths

        var chart = new d3plus.Geomap()
            .topojson(JSON.parse(w.data.map))
            .data(data)
            .legend(false)
            .ocean('transparent')
            .tiles(false)
            .zoom(false)
            .config({
                shapeConfig: {
                    Path: {
                        ariaLabel: (d, i) => "my"
                    }
                }
            })
            .select(cbody).render();

Tried with both d3plus-geomap.full.min.js and d3plus-geomap.full.js from the latest version link

**Comment by yemoku** [comment]
A small sample for your perusal if you have the time
[FoodCosts.json 20220718 2332.zip](https://github.com/d3plus/d3plus-geomap/files/9136095/FoodCosts.json.20220718.2332.zip)

Thanks

**Comment by davelandry** [comment]
@yemoku looking at your sample code, I can see that the `ariaLabel` attribute is working correctly (all `<path>` elements are getting their `aria-label` attribute set). The `aria-label` attribute is generally used for screen readers and assistive technologies to make the visualizations more accessible.

Screen Shot 2022-07-19 at 11 20 37 AM

My hunch is that you are misunderstanding the function of `ariaLabel`, and are looking for actual _visible_ labels that appear for the shapes? If so, then you are missing a config method that links the `data` Array IDs with the unique IDs in your `topojson` data. In your `topojson` data, it looks like you are trying to map States, and those unique State names are nested inside of the `properties` Object of each shape. Add these 3 config methods to link your data to the topojson, color the map by number of projects, and show the number in the tooltips:

```js
.topojsonId(d => d.properties.State)
.colorScale("Projects")
.tooltipConfig({
    tbody: [
        [d => `# of Projects: ${d.Projects}`]
    ]
})
```

**Comment by yemoku** [comment]
Yes, you are correct about my confusion on aria as opposed to visible labels. 

So the config changes you suggested work for a tooltip. Has there even a config to place actual labels/text on each shape (state in this example) in order to provide the (state0 names?

Thanks

**Comment by davelandry** [comment]
There is currently no default/shortcut for Geomap labels, but I would use the [`largestRect`](https://github.com/d3plus/d3plus-shape#largestRect) function in conjunction with the `labelBounds` method, which allows you to specify the bounding box of a shape's label:

```js
.shapeConfig({
    label: d => d.id,
    labelBounds: (d, i, s) => {

        const poly = d3plus.largestRect(s.points);

        return poly ? {
            x: poly.cx - poly.width / 2,
            y: poly.cy - poly.height / 2,
            width: poly.width,
            height: poly.height
        } : null;

    }
})
```

**Comment by davelandry** [comment]
https://github.com/d3plus/d3plus/issues/754

**Comment by yemoku** [comment]
Thanks immensely for this. Will give it a try

**Pull request merged**